lib/ClassMover/Adapter/TolerantParser/TolerantClassReplacer.php

```diff
@@ -34,6 +34,18 @@ public function replaceReferences(
 
         foreach ($classRefList as $classRef) {
             assert($classRef instanceof ClassReference);
+
+            // Check if the class name is fully qualified and replace it with a fully qualified version of the new name
+            $firstClassNameCharacter = $source->__toString()[$classRef->position()->start()];
```

Comment:
I think that this is probably not the right way to check if a class name is fully qualified or not. Any suggestions?

Reply:
maybe add `NodeUtil::isFullyQualified($name)` as with:

https://github.com/phpactor/phpactor/blob/1b10fbb0042d5f25ff59f39b94981c8ba76e3739/lib/Name/NameUtil.php#L34

Reply:
`$classRef instanceof FullyQualifiedName` ?

Reply:
looks like you could add a method to `QualifiedName`: `wasFullyQualified`. ATM it seems it then has the contents `["", "Foo", "Bar"]` otherwise you could check if it was fully qualified in the static constructor.

Reply:
This should work but the ClassMover namespace has it's own version of a FullyQualified name. Should I try to also add it to that? Because this changes a lot of tests if I do.

Reply:
Yes , class mover has it's own stuff...

Reply:
I've added it in both versions.

Reply:
AFAIK ClassMover doesn't use the `Name` package. Can we revert those changes? A name doesn't normally know if it _was_ fully qualfiied ,,,,

Reply:
Sure thing.